light-program-test/src/indexer/test_indexer.rs

```diff
@@ -0,0 +1,1272 @@
+    //                         },
+    //                     };
+    //                     compressed_accounts.push(compressed_account.clone());
+    //                     self.compressed_accounts.insert(0, compressed_account);
+    //                 }
+    //             }
+    //             None => {
+    //                 let compressed_account = CompressedAccountWithMerkleContext {
+    //                     compressed_account: compressed_account.compressed_account.clone(),
+    //                     merkle_context: MerkleContext {
+    //                         leaf_index: event.output_leaf_indices[i],
+    //                         merkle_tree_pubkey: event.pubkey_array
+    //                             [event.output_compressed_accounts[i].merkle_tree_index as usize],
+    //                         nullifier_queue_pubkey,
+    //                         queue_index: None,
+    //                     },
+    //                 };
+    //                 compressed_accounts.push(compressed_account.clone());
+    //                 self.compressed_accounts.insert(0, compressed_account);
+    //             }
+    //         };
+    //         let merkle_tree = &mut self
+    //             .state_merkle_trees
+    //             .iter_mut()
+    //             .find(|x| {
+    //                 x.accounts.merkle_tree
+    //                     == event.pubkey_array
+    //                         [event.output_compressed_accounts[i].merkle_tree_index as usize]
+    //             })
+    //             .unwrap()
+    //             .merkle_tree;
+    //         merkle_tree
+    //             .append(
+    //                 &compressed_account
+    //                     .compressed_account
+    //                     .hash::<Poseidon>(
+    //                         &event.pubkey_array
+    //                             [event.output_compressed_accounts[i].merkle_tree_index as usize],
+    //                         &event.output_leaf_indices[i],
+    //                     )
+    //                     .unwrap(),
+    //             )
+    //             .expect("insert failed");
+    //     }
+    //
+    //     self.events.push(event.clone());
+    //     (compressed_accounts, token_compressed_accounts)
+    // }
+
+    async fn create_proof_for_compressed_accounts(
+        &mut self,
+        compressed_accounts: Option<&[[u8; 32]]>,
+        state_merkle_tree_pubkeys: Option<&[Pubkey]>,
+        new_addresses: Option<&[[u8; 32]]>,
+        address_merkle_tree_pubkeys: Option<Vec<Pubkey>>,
+        rpc: &mut R,
+    ) -> ProofRpcResult {
+        if compressed_accounts.is_some()
+            && ![1usize, 2usize, 3usize, 4usize, 8usize]
+                .contains(&compressed_accounts.unwrap().len())
+        {
+            panic!(
+                "compressed_accounts must be of length 1, 2, 3, 4 or 8 != {}",
+                compressed_accounts.unwrap().len()
+            )
```

Comment:
we should remove the panics (we have these in multiple places including test indexer and test utils because I was lazy).

Reply:
It's probably a good time to do it in this pr since we are defining a new trait hence let's make the trait return results instead of having functions panic.